web/source/kmwdom.ts

```diff
@@ -82,6 +101,64 @@ namespace com.keyman {
       }
     }
 
+    /**
+     * Function     beep          KB      (DOM-side implementation)
+     * Scope        Public
+     * @param       {Object}      Pelem     element to flash
+     * Description  Flash body as substitute for audible beep; notify embedded device to vibrate
+     */    
+    doBeep(outputTarget: text.OutputTarget) {
+      // Handles embedded-mode beeps.
+      let keyman = com.keyman.singleton;
+      if ('beepKeyboard' in keyman) {
+        keyman['beepKeyboard']();
+      }
+
+      // All code after this point is DOM-based, triggered by the beep.
+      var Pelem: HTMLElement = outputTarget.getElement();
+      if(outputTarget instanceof dom.DesignIFrame) {
+        Pelem = outputTarget.docRoot; // I1446 - beep sometimes fails to flash when using OSK and rich control
+      }
+
+      if(!Pelem) {
+        return; // There's no way to signal a 'beep' to null, so just cut everything short.
+      }
+      
+      if(!Pelem.style || typeof(Pelem.style.backgroundColor)=='undefined') {
+        return;
+      }
+
+      for(var Lbo=0; Lbo<this._BeepObjects.length; Lbo++) { // I1446 - beep sometimes fails to return background color to normal
+                                                                  // I1511 - array prototype extended
+        if(this._BeepObjects[Lbo].e == Pelem) {
+          return;
+        }
+      }
+      
+      this._BeepObjects = com.keyman.singleton._push(this._BeepObjects, new BeepData(Pelem));
+      Pelem.style.backgroundColor = '#000000';
```

Comment:
This is probably not the time for this, but of course if we are in 'dark mode', the beep will not be very visible!

Reply:
I've gone ahead and made an issue for it.  I set it to "future" for now, since it seems like a low-priority item.